### PR TITLE
interval for rect

### PR DESCRIPTION
**Pull request by mbostock**
Adds an _interval_ option for rects which transforms _x_ to _x1_ and _x2_ using _interval_.floor and _interval_.offset. This provides a suitable replacement for bars when an axis is temporal rather than ordinal. Related #513, and perhaps this could be considered a fix.

I’d further love to be able to specify a named interval such as “day” or “hour”, but this requires knowing the time zone (or for now, whether the associated scale is “utc” or local “time”). That’s not easily accessible when evaluating a transform since this happens before scales are constructed. Although I suspect that with a top-level timeZone option #531 we could find some way of passing this down to the transform. At any rate, convenience shorthand can be added in the future.

### CODE REVIEW

**Comment by mbostock** [comment]
I guess this should probably set default insets, too, like e.g. bin does?

**Review by Fil — COMMENTED** [review]
Should we expose this in some way, so that custom marks could have access to this new transform?
I'm sending a few suggestions in another PR

File: src/transforms/inset.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export function maybeInsetY({inset, insetTop, insetBottom, ...options} = {}) {
+  ([insetTop, insetBottom] = maybeInset(inset, insetTop, insetBottom));
+  return {inset, insetTop, insetBottom, ...options};
+}
+
+function maybeInset(inset, inset1, inset2) {
+  return inset === undefined && inset1 === undefined && inset2 === undefined
+    ? (offset ? [1, 0] : [0.5, 0.5])
```
[the remark this answers is not in the document]
Comment (mbostock) [reply src/transforms/inset.js:15]: I’m aware of this problem but not going to fix it here. 👍

**Comment by mbostock** [comment]
I don’t think we should expose the interval transform yet, out of conservatism, but we can always chose to do so in the future if we think it’s useful.

**Comment by mbostock** [comment]
I think I want to add the interval transform to bar and rule, too.

**Comment by mbostock** [comment]
Added. Works nicely!

**Review by Fil — APPROVED** [review]

**Pull request merged**